### DISCUSSION TITLE
[Features request] qbittorrent "Category" and its settings.

### DISCUSSION THREAD

**Discussion by lilws** [jesec/flood#181] [General]
In qbittorrent, they introduce Category, work like a tag but its features is "Automatic torrent management". It mainly focus on torrent management for saving location. User can define new category and default saving location for that category. It would be nice if flood can show "Category" rather than "Tag" and allow user to define new category and save location. And enable the Automatic torrent management from right click context menu.

**Comment by jesec** [comment]
No. There is absolutely no plan to introduce such inconsistency. Additionally, qBittorrent is weighing to remove the feature (basically ditching categories in favor of tags) and only allow save path assignment on add in PR 13773.

Flood already has per-tag preferred download destination, which is automatically applied when a torrent is added.

**Reply by lilws** [reply]
Thank you for let me know it. In qbittorrent github it has like 2.3k issues, I didn't know that they are going to ditch categories. Although it a nice feature so far, In rtorrent they have feature auto add tag in watch folder. Hope qbittorrent will implement that soon. Missed that features.

**Reply by C84186** [reply]
@lilws  Apparently they're keeping categories, but removing some of the save path management stuff

**Comment by C84186** [comment]
As @jesec is adamant he will not incorporate categories into his project, which is his prerogative, I've hacked together a workaround script in my own repo. 

https://github.com/C84186/qBittorrent_tag_by_pattern/blob/main/synchronize_cat_tags.py 

It's a bit clunky, but essentially gives you a get/set pattern for managing categories. 

If you set that script up to run regularly, say, every 10ish minutes, it should be sufficient to  meet your needs if you're really tied to the idea of categories.

Basically, when you want to assign a category, eg, "Movies" you tag the torrent with a set tag `_SET_Movies`. 

The script will categories the torrent as "Movies", then clear the set flags.

The script will automatically tag anything in a given cat, eg, Movies, as "_CAT_Movies" , and make sure that there's only one _CAT_ tag associated with a given torrent

**Reply by C84186** [reply]
https://github.com/C84186/qBittorrent_tag_by_pattern/blob/main/compose.yml I've dockerized this now, you should just be able to add something like this to your compose file & your cats should stay synched.
